<a href="https://colab.research.google.com/github/ARizalma/river-water-quality-analysis/blob/main/notebooks/01_data_understanding.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# River Water Quality Analysis

## Notebook 01 — Data Understanding

In [14]:
# Import Libraries

import pandas as pd
import numpy as np

In [15]:
# Load Dataset

import requests

url = "https://raw.githubusercontent.com/ARizalma/river-water-quality-analysis/main/Data/river-water-quality-data-for-2023.csv"

df = pd.read_csv(url, sep=";")

In [16]:
## Data Overview

df.head()

,titik_sampel,lintang_selatan,bujur_timur,jenis_parameter,parameter,baku_mutu,hasil_pengukuran,periode_data,periode_pemantauan,bulan_sampling,nama_sungai,alamat
0,ANK-1,"-6,175882376","106,7286948",Fisika,TDS,1000,164,2023,Periode 1 (Musim Hujan),2,Angke,Jl. Duri Kosambi (Ciledug)
1,ANK-1,"-6,175882376","106,7286948",FIsika,TSS,50,24,2023,Periode 1 (Musim Hujan),2,Angke,Jl. Duri Kosambi (Ciledug)
2,ANK-1,"-6,175882376","106,7286948",Kimia,pH,6-9,"7,56",2023,Periode 1 (Musim Hujan),2,Angke,Jl. Duri Kosambi (Ciledug)
3,ANK-1,"-6,175882376","106,7286948",Kimia,BOD,3,"23,65",2023,Periode 1 (Musim Hujan),2,Angke,Jl. Duri Kosambi (Ciledug)
4,ANK-1,"-6,175882376","106,7286948",Kimia,COD,25,189,2023,Periode 1 (Musim Hujan),2,Angke,Jl. Duri Kosambi (Ciledug)


## Dataset Dimensions

In [17]:
df.shape

(10911, 12)

## Column Names

In [18]:
df.columns

Index(['titik_sampel', 'lintang_selatan', 'bujur_timur', 'jenis_parameter',
       'parameter', 'baku_mutu', 'hasil_pengukuran', 'periode_data',
       'periode_pemantauan', 'bulan_sampling', 'nama_sungai', 'alamat'],
      dtype='object')

##Dataset Information

In [19]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10911 entries, 0 to 10910
Data columns (total 12 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   titik_sampel        10911 non-null  object
 1   lintang_selatan     10911 non-null  object
 2   bujur_timur         10911 non-null  object
 3   jenis_parameter     10911 non-null  object
 4   parameter           10911 non-null  object
 5   baku_mutu           10911 non-null  object
 6   hasil_pengukuran    10911 non-null  object
 7   periode_data        10911 non-null  int64 
 8   periode_pemantauan  10911 non-null  object
 9   bulan_sampling      10911 non-null  int64 
 10  nama_sungai         10911 non-null  object
 11  alamat              10911 non-null  object
dtypes: int64(2), object(10)
memory usage: 1023.0+ KB


In [20]:
df[['hasil_pengukuran', 'baku_mutu']].sample(10)

,hasil_pengukuran,baku_mutu
2491,"0,0029","0,05"
9069,"0,0029","0,05"
5199,"0,025","0,03"
2206,"0,18","0,2"
2695,"0,0069","0,05"
7261,"2,51",3
65,"0,48","0,2"
5147,"3,45",4
10132,14000000,1000
6893,"9,67",3


In [21]:
df['hasil_pengukuran'].nunique()

2014

In [22]:
df['jenis_parameter'].value_counts()

,count
jenis_parameter,
Kimia,9006
Biologi,946
Fisika,480
FIsika,479


## Descriptive Statistics

In [23]:
df.describe()

,periode_data,bulan_sampling
count,10911.0,10911.000000
mean,2023.0,6.011823
std,0.0,2.752708
min,2023.0,2.000000
25%,2023.0,2.000000
50%,2023.0,8.000000
75%,2023.0,9.000000
max,2023.0,9.000000


## Data Quality Assessment
### Missing Values

In [24]:
df.isnull().sum()

,0
titik_sampel,0
lintang_selatan,0
bujur_timur,0
jenis_parameter,0
parameter,0
baku_mutu,0
hasil_pengukuran,0
periode_data,0
periode_pemantauan,0
bulan_sampling,0


### Duplicate Records

In [25]:
df.duplicated().sum()

np.int64(0)

###Data Type Issues

In [37]:
df.dtypes

,0
titik_sampel,object
lintang_selatan,object
bujur_timur,object
jenis_parameter,object
parameter,object
baku_mutu,object
hasil_pengukuran,object
periode_data,int64
periode_pemantauan,object
bulan_sampling,int64


## Explore Key Variables
### Jenis Parameter

In [26]:
df['jenis_parameter'].value_counts()

,count
jenis_parameter,
Kimia,9006
Biologi,946
Fisika,480
FIsika,479


### Water Quality Parameters

In [27]:
df['parameter'].value_counts()

,count
parameter,
TDS,480
pH,480
BOD,480
COD,480
Total P,480
DO,479
TSS,479
MBAS,479
Fenol,478


###River Names

In [29]:
df['nama_sungai'].value_counts()

,count
nama_sungai,
Ciliwung,1656
Kalibaru Timur,920
Sunter,736
Cipinang,644
Cideng,556
Kalibaru Barat,552
Mampang,552
Grogol,552
Cakung,490


###Monitoring Period

In [32]:
df['periode_pemantauan'].value_counts().sort_index()

,count
periode_pemantauan,
Periode 1 (Musim Hujan),2760
Periode 2 (Musim Peralihan),2631
Periode 3 (Musim Kemarau),2760
Periode 4 (Musim Peralihan),2760


###Sampling Month

In [33]:
df['bulan_sampling'].value_counts().sort_index()

,count
bulan_sampling,
2,2760
5,2631
8,2760
9,2760


###Measurement Values

In [35]:
df[['hasil_pengukuran','baku_mutu']].sample(10,random_state=50)

,hasil_pengukuran,baku_mutu
6569,28,50
7786,3000000,1000
6001,"0,6","0,2"
9796,"0,21","0,2"
3597,"0,06","0,2"
10246,9000000,5000
8792,"0,019","0,03"
10597,354,25
8629,"0,0069","0,05"
7404,"0,5","0,2"


## Initial Findings

### Observations

- The dataset contains 10,911 observations and 12 variables.
- No missing values or duplicate records were found.
- The dataset consists of physical, chemical, and biological water quality parameters.
- The `hasil_pengukuran` column stores decimal values using commas (`,`), causing the data type to be detected as `object`.
- Inconsistent category labels were identified in the `jenis_parameter` column (e.g., `Fisika` and `FIsika`).
- Several columns require data type conversion and cleaning before exploratory data analysis.